### PR TITLE
[Doc] Improve Input docs

### PR DESCRIPTION
**Pull request by fzaninotto**

### CODE REVIEW

File: docs/RichTextInput.md [hunk 1]
```diff
@@ -155,3 +113,57 @@ export const MyEditorOptions = {
 	],
 };
 ```
+
+## `toolbar`
+
+The `<RichTextInput>` component has a `toolbar` prop that accepts a `ReactNode`.
+
+You can leverage this to change the buttons [size](#api):
```
Comment (WiXSL) [inline docs/RichTextInput.md:121]: Wrong anchor: ... `[size]`(#api): 
Could be `https://mui.com/material-ui/api/button` ?

**Review by WiXSL — CHANGES_REQUESTED** [review]

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/RichTextInput.md (hunk 1 above)
Comment (fzaninotto) [reply docs/RichTextInput.md]: You're right, I removed the anchor.

File: docs/DateInput.md [hunk 2]
```diff
@@ -5,20 +5,41 @@ title: "The DateInput Component"
...
+## Usage
+
 ```jsx
 import { DateInput } from 'react-admin';
 
 <DateInput source="published_at" />
 ```
 
-`<DateInput>` also accepts the [common input props](./Inputs.md#common-input-props).
+The field value must be a string with the pattern `YYYY-MM-DD` (ISO 8601), e.g. `'2022-04-30'`.
```
Comment (slax57) [inline docs/DateInput.md:22]: I believe it also accepts JS `Date` objects, doesn't it? (even though in this case it requires a custom `parse` function)

File: docs/DateTimeInput.md [hunk 3]
```diff
@@ -5,16 +5,27 @@ title: "The DateTimeInput Component"
...
+## Usage
+
 ```jsx
 import { DateTimeInput } from 'react-admin';
 
 <DateTimeInput source="published_at" />
 ```
 
-`<DateTimeInput>` also accepts the [common input props](./Inputs.md#common-input-props).
+The input value must be a valid date string, i.e. a string understood by JavasSript's [`Date.parse()` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse). Strings with [the ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601) 'yyyy-MM-ddThh:mm' are the most common (e.g. `'2022-04-30T12:30'`). The field value may contain a timezone offset, e.g. `'2022-04-30T12:30+02:00'`. If no timezone is specified, the browser's timezone is used.
```
Comment (slax57) [inline docs/DateTimeInput.md:22]: Same, I believe it also accepts `Date` objects

File: docs/FileInput.md [hunk 4]
```diff
@@ -5,56 +5,165 @@ title: "The FileInput Component"
...
+
+```jsx
+import { FileInput, FileField } from 'react-admin';
+
+<FileInput source="attachments">
+    <FileField source="src" title="title" />
+</FileInput>
+```
+
+`<FileInput>` uses its child component to give a preview of the files. `<FileInput>` renders it child once per file, inside a `<RecordContext>`, so the child can be a Field component. The default [`<FileField>`](./FileField.md) renders the name of the file(s), with a hyperlink.
```
Comment (slax57) [inline docs/FileInput.md:24]:
```suggestion
`<FileInput>` uses its child component to give a preview of the files. `<FileInput>` renders its child once per file, inside a `<RecordContext>`, so the child can be a Field component. The default [`<FileField>`](./FileField.md) renders the name of the file(s), with a hyperlink.
```

File: docs/ImageInput.md [hunk 5]
```diff
@@ -5,53 +5,149 @@ title: "The ImageInput Component"
 
 # `<ImageInput>`
 
-`<ImageInput>` allows to upload some pictures using [react-dropzone](https://github.com/okonet/react-dropzone).
+`<ImageInput>` allows editing and uploading iumages (png, jpg, gif, etc). It is powered by [react-dropzone](https://github.com/okonet/react-dropzone).
```
Comment (slax57) [inline docs/ImageInput.md:8]:
```suggestion
`<ImageInput>` allows editing and uploading images (png, jpg, gif, etc). It is powered by [react-dropzone](https://github.com/okonet/react-dropzone).
```

File: docs/ImageInput.md [hunk 6]
```diff
@@ -5,53 +5,149 @@ title: "The ImageInput Component"
...
+
+```jsx
+import { ImageInput, ImageField } from 'react-admin';
+
+<ImageInput source="pictures" label="Related pictures">
+    <ImageField source="src" title="title" />
+</ImageInput>
+```
+
+`<ImageInput>` uses its child component to give a preview of the files. `<ImageInput>` renders it child once per file, inside a `<RecordContext>`, so the child can be a Field component. The default [`<ImageField>`](./ImageField.md) renders the name of the file(s), with a hyperlink.
```
Comment (slax57) [inline docs/ImageInput.md:22]: don't think there is an hyperlink, only the image is displayed I believe

File: docs/ImageInput.md [hunk 7]
```diff
@@ -5,53 +5,149 @@ title: "The ImageInput Component"
...
+- '.png,.gif,.jpg,.jpeg' 
+- 'image/*'
 
-`<ImageInput>` delegates the preview of currently selected images to its child. `<ImageInput>` clones its child as many times as there are selected images, passing the image as the `record` prop. To preview a simple list of image thumbnails, you can use `<ImageField>` as child, as follows:
+If left empty, all file types are accepted (even non-image types).
+
+## `children`
+
+`<ImageInput>` delegates the preview of currently selected images to its child. `<ImageInput>` clones its child component once per file, inside a [`<RecordContext>`](./useRecordContext.md), so the child can be a [Field](./Fields.md) component. The default [`<ImageField>`](./ImageField.md) renders the name of the file(s), with a hyperlink.
```
Comment (slax57) [inline docs/ImageInput.md:86]: same

**Review by slax57 — CHANGES_REQUESTED** [review]

File: docs/DateInput.md (hunk 2 above)
Comment (fzaninotto) [reply docs/DateInput.md:22]: Yes, but I don't see the point in mentioning it, as the data comes from an API response - and therefore was serialized to string. Unless some data providers return dates as `Date` objects?

File: docs/ImageInput.md (hunk 6 above)
Comment (fzaninotto) [reply docs/ImageInput.md]: Right, fixed

File: docs/ImageInput.md (hunk 7 above)
Comment (fzaninotto) [reply docs/ImageInput.md]: Fixed

File: docs/DateTimeInput.md (hunk 3 above)
Comment (fzaninotto) [reply docs/DateTimeInput.md]: Here, I think it makes sense to mention it, so I modified the doc.

**Review by WiXSL — APPROVED** [review]

File: docs/DateInput.md (hunk 2 above)
Comment (slax57) [reply docs/DateInput.md:22]: Good point. I don't know any dataProvider that do that, my comment was based on the code only.

**Review by slax57 — APPROVED** [review]

**Pull request merged**